packages/bot-skeleton/src/scratch/dbot.js

```diff
@@ -22,7 +22,7 @@ class DBot {
     /**
      * Initialises the workspace and mounts it to a container element (app_contents).
      */
-    async initWorkspace(public_path, store, api_helpers_store) {
+    async initWorkspace(public_path, store, api_helpers_store, ui) {
```

Comment:
I suggest passing in an `is_mobile` flag rather than the entire `ui` store, consider that the mobile app won't have a UI store.